tokio-postgres/src/query.rs

```diff
@@ -242,3 +242,13 @@ impl RowStream {
         self.rows_affected
     }
 }
+
+pub async fn sync(client: &InnerClient) -> Result<(), Error> {
+    let buf = Bytes::from_static(b"S\0\0\0\x04");
```

Comment:
Naive question, but what does this message mean? I mean, I guess it's SYNC looking at your upstream PR, but what exactly it does?

Reply:
Correct, it's the "Sync" message. In the extended query protocol, you send several messages to execute a query, usually "Parse, Describe, Bind, Execute, Sync". But the protocol allows you to mix and match, so you can bind/execute multiple times before issuing the sync. Sync is the implicit commit point.

Conveniently, it works without sending anything. So a Sync on it's own just means "commit nothing". It skips any query parsing or execution steps.

This forms the cheapest ping-pong that exists in the postgres protocol, cheaper than a "SELECT 1", and given we have audit logging, it doesn't show in the audit log.